### ISSUE TITLE
Multithreaded BagOfVisualWords: "memory corrupt" problem

### ISSUE THREAD

**Issue by vkpankov** [accord-net/framework#549]
```C#
var Bow = BagOfVisualWords.Create(500);
Bow.ParallelOptions.MaxDegreeOfParallelism = 8;
Bow.Learn(TrainImages.ToArray());

```
`TrainImages` read from file: 
```C#
Bitmap image = ((Bitmap)Accord.Imaging.Image.FromFile(imageFileName));
```
Error: attempted read or write to protected memory

If I set `MaxDegreeOfParallelism=1`, it's all OK.

**Comment by blaisexen** [comment]
listening....

**Comment by cesarsouza** [comment]
This should have been addressed in release 3.6.0. However, I am not completely sure it was completely resolved since I am not getting this memory corrupt problem quite often.
